geode-core/src/test/java/org/apache/geode/internal/cache/GemFireCacheImplTest.java

```diff
@@ -620,6 +625,40 @@ public void getCacheServers_isCanonical() {
         .isSameAs(gemFireCacheImpl.getCacheServers());
   }
 
+  @Test
+  public void testLockDiskStore() throws InterruptedException {
+    int nThread = 10;
+    String diskStoreName = "MyDiskStore";
+    AtomicInteger nTrue = new AtomicInteger();
+    AtomicInteger nFalse = new AtomicInteger();
+    ExecutorService executorService = Executors.newFixedThreadPool(nThread);
```

Comment:
I had thought about using `ExecutorServiceRule`. However, the return value of `doLockDiskStore()` is non-deterministic. I can't assert the return value of individual thread. I can only count and assert the number of returning values when all threads are done. I am not sure how `ExecutorServiceRule` will serve this test case.

Reply:
I recommend using ExecutorServiceRule for several reasons. Most important is that it always performs tearDown (ie shutdownNow) even if the test blows up before invoking `executorService.shutdown();`

Reply:
Another alternative is to extract your `executorService` as a field so that shutdown or shutdownNow is always invoked even if the test has a failure:
```
private ExecutorService executorService;

@Before
public void setUp() {
  executorService = Executors.newCachedThreadPool();
}

@After
public void tearDown() {
  executorService.shutdownNow();
}
```

Reply:
I have changed new test in `GemFireCacheImplTest` to use `ExecutorServiceRule`.